program-runtime/src/invoke_context.rs

```diff
@@ -815,6 +817,17 @@ impl<'a> InvokeContext for ThisInvokeContext<'a> {
             .map(|frame| &frame.keyed_accounts[frame.keyed_accounts_range.clone()])
             .ok_or(InstructionError::CallDepth)
     }
+    fn get_instruction_keyed_accounts(&self) -> Result<&[KeyedAccount], InstructionError> {
+        let frame = self
+            .invoke_stack
+            .last()
+            .ok_or(InstructionError::CallDepth)?;
+        let first_instruction_account = frame
+            .number_of_program_accounts
+            .checked_sub(1)
```

Comment:
@Lichtso why do you subtract 1 here?

Reply:
Because the `number_of_program_accounts - 1` is `first_instruction_account`.
Or in other words `number_of_program_accounts` is `first_instruction_account + 1`.

`first_instruction_account` is actually the index of the `program_id` of that instruction, see:
https://github.com/solana-labs/solana/blob/c9bfc99c720ed091c326c1cce58fcd8197306235/program-runtime/src/invoke_context.rs#L132